target/ppc/translate/vmx-impl.c.inc

```diff
@@ -1491,6 +1491,42 @@ static bool trans_VSRDBI(DisasContext *ctx, arg_VN *a)
     return true;
 }
 
+static bool do_vcntmb(DisasContext *ctx, arg_VX_mp *a, int vece)
+{
+    TCGv_i64 rt, vrb, mask;
+
+    REQUIRE_INSNS_FLAGS2(ctx, ISA310);
+    REQUIRE_VECTOR(ctx);
+
+    rt = tcg_const_i64(0);
+    vrb = tcg_temp_new_i64();
+    mask = tcg_constant_i64(dup_const(vece, 1ULL << ((8 << vece) - 1)));
+
+    for (int i = 0; i < 2; i++) {
+        get_avr64(vrb, a->vrb, i);
+        if (a->mp) {
+            tcg_gen_and_i64(vrb, mask, vrb);
+        } else {
+            tcg_gen_andc_i64(vrb, mask, vrb);
+        }
+        tcg_gen_ctpop_i64(vrb, vrb);
+        tcg_gen_add_i64(rt, rt, vrb);
+    }
+
+    tcg_gen_shli_i64(rt, rt, TARGET_LONG_BITS - 8 + vece);
```

Comment:
Theoretically, it is not 64-bit-only since PowerISA doesn't say so explicitly. But in practice, the only processor that implements these instructions is 64-bit. With cfuged/vcfuged and other similar cases, we kept the vector version available to a possible 32-bit implementation of PowerISA v3.1, so I did the same here.

Reply:
I'm not necessarily disagreeing with keeping it also for 32-bit. The ISA is confusing, though, and it's not clear to me that a 32-bit version needs to be supported.

Reply:
The instruction description could be understood to 32-bits, since the result is paced in the highers bits, it's just the zeroing that wouldn't make sense. The pseudo code is a bit more problematic, but that's not exclusive to vcntmb... Anyway, the extra temporary and truncating shouldn't have much of an impact, so I think I'll keep it like that for now.